src/Core/Install.php

```diff
@@ -122,34 +163,37 @@ private static function addCheck(&$checks, $title, $status, $required, $help, $e
 	 * - Checks if it is the CLI version
 	 * - Checks if "register_argc_argv" is enabled
  	 *
-	 * @param string $phpath Optional. The Path to the PHP-Binary
-	 * @param array $checks The list of all checks (by-ref parameter!)
+	 * @param string $phppath Optional. The Path to the PHP-Binary
+	 * @param bool   $required Optional. If set to true, the PHP-Binary has to exist (Default false)
+	 *
+	 * @return bool false if something required failed
 	 */
-	public static function checkPHP($phpath, &$checks)
+	public function checkPHP($phppath = null, $required = false)
 	{
 		$passed = $passed2 = $passed3 = false;
-		if (strlen($phpath)) {
-			$passed = file_exists($phpath);
+		if (isset($phppath)) {
+			$passed = file_exists($phppath);
 		} else {
-			$phpath = trim(shell_exec('which php'));
-			$passed = strlen($phpath);
+			$phppath = trim(shell_exec('which php'));
+			$passed = strlen($phppath);
 		}
+
 		$help = "";
 		if (!$passed) {
 			$help .= L10n::t('Could not find a command line version of PHP in the web server PATH.') . EOL;
 			$help .= L10n::t("If you don't have a command line version of PHP installed on your server, you will not be able to run the background processing. See <a href='https://github.com/friendica/friendica/blob/master/doc/Install.md#set-up-the-worker'>'Setup the worker'</a>") . EOL;
 			$help .= EOL . EOL;
 			$tpl = get_markup_template('field_input.tpl');
 			$help .= replace_macros($tpl, [
-				'$field' => ['phpath', L10n::t('PHP executable path'), $phpath, L10n::t('Enter full path to php executable. You can leave this blank to continue the installation.')],
+				'$field' => ['phpath', L10n::t('PHP executable path'), $phppath, L10n::t('Enter full path to php executable. You can leave this blank to continue the installation.')],
 			]);
-			$phpath = "";
+			$phppath = "";
 		}
 
-		self::addCheck($checks, L10n::t('Command line PHP').($passed?" (<tt>$phpath</tt>)":""), $passed, false, $help);
+		self::addCheck(L10n::t('Command line PHP') . ($passed ? " (<tt>$phppath</tt>)" : ""), $passed, false, $help);
```

Comment:
Shouldn't it be `$this->addCheck` instead?

Reply:
Good to know that PHP is auto-using the `$this->addCheck` automatically ;-) .. Because I saw the checks in my debug output